clib/src/ser.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright © 2021 The Radicle Link Contributors
+//
+// This file is part of radicle-link, distributed under the GPLv3 with Radicle
+// Linking Exception. For full terms see the included LICENSE file.
+
+use thiserror::Error;
+
+use minicbor::Encode;
+use serde::Serialize;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error(transparent)]
+    Json(#[from] serde_json::Error),
+    #[error(transparent)]
+    Cbor(#[from] minicbor::encode::Error<std::io::Error>),
+    #[error(transparent)]
+    Utf8(#[from] std::string::FromUtf8Error),
+}
+
+/// An enumeration of the formats the CLI can output. Note that since any of
+/// these formats can be used, the corresponding data type needs to implement
+/// the required traits.
+pub enum Format {
+    /// Requires the data type to implement [`Serialize`].
+    Json,
+    /// Requires the data type to implement [`Encode`].
+    Cbor,
+}
+
+impl Format {
+    /// Serialize the `val` to a `String`.
+    pub fn to_string<T>(&self, val: &T) -> Result<String, Error>
```

Comment:
This feels (to me) confusingly named because of the ubiquity of `ToString::to_string`, would `format` be better?

Reply:
You're right, I was just cargo culting with `serde_json::to_string` being the naming. I like the `format` suggestion, thanks :)